### PR TITLE
Work around awkward SO_TIMESTAMP definition.

### LINKED ISSUE #1861: Use of unresolved identifier 'SO_TIMESTAMP'
I'm attempting to recompile a Swift package on my Raspberry Pi 4 originally built with swift-nio-ssl 2.10.0 and now I'm getting multiple errors along the lines of:

.../.build/checkouts/swift-nio/Sources/NIO/BSDSocketAPI.swift:404:43: error: use of unresolved identifier 'SO_TIMESTAMP'
            NIOBSDSocket.Option(rawValue: SO_TIMESTAMP)

### SwiftNIO version/commit hash
swift-nio-ssl.git at 2.13.1

### Swift & OS version (output of `swift --version && uname -a`)
Swift version 5.1.5 (swift-5.1.5-RELEASE)
Target: armv6-unknown-linux-gnueabihf
Linux pi4 5.10.17-v7l+ #1414 SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

On some Linux platforms the definition of SO_TIMESTAMP is too complex
for the clang importer to handle. Delegate to C to find out what it
should be.

Modifications:

- Define SO_TIMESTAMP in CNIOLinux instead.

Result:

Should build on the awkward Linux platforms.
Resolves #1861

### CODE REVIEW

**Comment by Lukasa** [comment]
@picnicbob Can you validate that this works for you?

**Review by glbrntt — APPROVED** [review]

**Comment by picnicbob** [comment]
@Lukasa I still have a build error for SO_RCVTIMEO

**Comment by Lukasa** [comment]
@picnicbob Can you provide the output here?

**Comment by picnicbob** [comment]
@Lukasa This is one of many for the same SO_RCVTIMEO constant:
`.../.build/checkouts/swift-nio/Sources/NIO/BSDSocketAPI.swift:393:43: error: use of unresolved identifier 'SO_RCVTIMEO'
            NIOBSDSocket.Option(rawValue: SO_RCVTIMEO)`

**Comment by Lukasa** [comment]
Ok, we've at least moved the bar here: now we're at `SO_RCVTIMEO`.

**Comment by Lukasa** [comment]
@picnicbob Try again?

**Comment by picnicbob** [comment]
@Lukasa Success!

**Comment by picnicbob** [comment]
@Lukasa Well... it compiled, but when I run I hit an Illegal Instruction in my call to `let allocator = AdaptiveRecvByteBufferAllocator()`
Is this related?

**Review by fabianfett — APPROVED** [review]
LGTM

**Review by glbrntt — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-nio-bot please test

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
Oh: https://www.githubstatus.com/incidents/76nv9h8pmkv4

File: Sources/CNIOLinux/shim.c [hunk 1]
```diff
@@ -143,4 +143,7 @@ size_t CNIOLinux_CMSG_LEN(size_t payloadSizeBytes) {
 size_t CNIOLinux_CMSG_SPACE(size_t payloadSizeBytes) {
     return CMSG_SPACE(payloadSizeBytes);
 }
+
+int CNIOLinux_SO_TIMESTAMP = SO_TIMESTAMP;
```
Comment (weissi) [inline Sources/CNIOLinux/shim.c:147]: @Lukasa these should be `const`, right?

Comment (Lukasa) [reply Sources/CNIOLinux/shim.c:147]: Yup, probably.

**Pull request merged**